packages/core/src/credential/Credential.ts

```diff
@@ -81,103 +271,163 @@ export function verifyAgainstCType(
 ): ICredential {
+  const { hashes: claimHashes, nonceMap: claimNonceMap } =
+    Claim.hashClaimContents(claim)
+
+  const rootHash = calculateRootHash({
+    legitimations,
+    claimHashes,
+    delegationId,
+  })
+
+  // signature will be added afterwards!
   const credential = {
-    request,
-    attestation,
+    claim,
+    legitimations,
+    claimHashes,
+    claimNonceMap,
+    rootHash,
+    delegationId,
   }
   verifyDataStructure(credential)
   return credential
 }
 
-/**
- * Custom Type Guard to determine input being of type ICredential.
- *
- * @param input The potentially only partial ICredential.
- *
- * @returns Boolean whether input is of type ICredential.
- */
-export function isICredential(input: unknown): input is ICredential {
-  try {
-    verifyDataStructure(input as ICredential)
-  } catch (error) {
-    return false
-  }
-  return true
+type VerifyOptions = {
+  ctype?: ICType
+  challenge?: string
+  resolver?: IDidResolver
+  checkSignatureWithoutChallenge?: boolean
 }
 
 /**
- * Verifies whether the credential is valid. It is valid if:
- * * the data is valid (see [[verifyDataIntegrity]]);
- * and
- * * the [[Attestation]] object for this credential is valid (see [[Attestation.checkValidity]], where the **chain** is queried).
+ * Verifies data structure, data integrity and claimers signature.
  *
  * Upon presentation of a credential, a verifier would call this [[verify]] function.
  *
- * @param credential - The credential to check for validity.
- * @param verificationOpts The additional options to use upon attested credential verification.
- * @param verificationOpts.resolver - The resolver used to resolve the claimer's identity if it is not passed in.
- * Defaults to [[DidResolver]].
- * @param verificationOpts.challenge - The expected value of the challenge. Verification will fail in case of a mismatch.
- * @returns A promise containing whether the provided credential is valid.
+ * @param credential - The object to check.
+ * @param options - Additional parameter for more verification steps.
+ * @param options.ctype - CType which the included claim should be checked against.
+ * @param options.challenge -  The expected value of the challenge. Verification will fail in case of a mismatch.
+ * @param options.checkSignatureWithoutChallenge - Wether to check the signature without a challenge being present.
+ * @param options.resolver - The resolver used to resolve the claimer's identity. Defaults to [[DidResolver]].
  */
 export async function verify(
   credential: ICredential,
-  verificationOpts: {
-    resolver?: IDidResolver
-    challenge?: string
-  } = {}
-): Promise<boolean> {
-  return (
-    verifyDataIntegrity(credential) &&
-    (await RequestForAttestation.verifySignature(
-      credential.request,
-      verificationOpts
-    )) &&
-    Attestation.checkValidity(credential.attestation)
-  )
+  {
+    ctype,
+    challenge,
+    resolver = DidResolver,
+    checkSignatureWithoutChallenge = true,
```

Comment:
isn't it safer If that defaults to false?

Reply:
as discussed further, I renamed the flag, since there might be usecases, where I am ok with an unsigned credential.